### PR TITLE
ISPN-11262 Remove deprecated eviction configuration values

### PR DESCRIPTION
**Pull request by wburns**
* EvictionStrategy removed
* Off heap address count removed

https://issues.redhat.com/browse/ISPN-11262

### CODE REVIEW

**Comment by ryanemerson** [comment]
Thanks @wburns. The wildfly integration also needs updating.

**Comment by wburns** [comment]
Weird, I am not sure why IntelliJ didn't inform me of the compilation errors.. thought master worked for all modules in IntelliJ now :(

**Comment by wburns** [comment]
Updated, just let me rebase when you are ready to integrate.

**Comment by ryanemerson** [comment]
@wburns Hibernate cache needs updating > 200 failures.

**Comment by wburns** [comment]
Test failures appear to be unrelated.

File: wildfly/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/InfinispanSubsystemXMLReader.java [hunk 1]
```diff
@@ -1619,7 +1621,12 @@ private void parseEviction(XMLExtendedStreamReader reader, ModelNode cache, Map<
             Attribute attribute = Attribute.forName(reader.getAttributeLocalName(i));
             switch (attribute) {
                 case STRATEGY: {
-                    enabled = EvictionStrategy.valueOf(value).isEnabled();
+                    try {
+                        enabled = EvictionStrategy.valueOf(value).isEnabled();
+                    } catch (IllegalArgumentException e) {
+                        // Unable to parse eviction strategy - assuming it is enabled
+                        enabled = true;
```
Comment (ryanemerson) [inline wildfly/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/InfinispanSubsystemXMLReader.java]: We should throw an Exception here if the parser is 11 or higher and the strategy is not recognised.

File: wildfly/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/InfinispanSubsystemXMLReader.java [hunk 2]
```diff
@@ -1592,7 +1592,9 @@ private void parseMemoryOffHeap(XMLExtendedStreamReader reader, ModelNode cache,
                     break;
                 }
                 case ADDRESS_COUNT: {
-                    MemoryOffHeapConfigurationResource.ADDRESS_COUNT.parseAndSetParameter(value, offHeap, reader);
+                    if (namespace.since(11, 0)) {
+                        throw ParseUtils.unexpectedElement(reader);
+                    }
```
Comment (ryanemerson) [inline wildfly/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/InfinispanSubsystemXMLReader.java]: We should warn that the attribute has been ignored via `ignoreAttribute`.

File: wildfly/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/InfinispanSubsystemXMLReader.java (hunk 1 above)
Comment (wburns) [reply wildfly/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/InfinispanSubsystemXMLReader.java]: Sure

File: wildfly/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/InfinispanSubsystemXMLReader.java (hunk 2 above)
Comment (wburns) [reply wildfly/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/InfinispanSubsystemXMLReader.java]: Sure

**Comment by wburns** [comment]
Fixed review comments.

**Comment by ryanemerson** [comment]
Thanks @wburns

**Pull request merged**